master/internal/stream/system_intg_test.go

```diff
@@ -246,6 +139,83 @@ func TestTrialStartup(t *testing.T) {
 	require.NoError(t, errgrp.Wait())
 }
 
+func buildStartupMsg(syncID, knownType, known string, expIDs, trialIDs []int) StartupMsg {
```

Comment:
question (non-blocking): how would this adjust for projects?

Reply:
good question - the `knownType` and `known` string can easily support projects, but the way known IDs are done is currently not extensible. I think the answer is to refactor the `expIDs` and `trialIDs` to be similar to known type and known - `subType` and `subIDs`. Thoughts?

Reply:
Would that mean that we can only subscribe to one type corresponding to the knownType? Another approach could be providing an `map[knownType][]interface` that way we can even accommodate for things that don't have traditional int ids